modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -237,6 +237,10 @@ const FrontendDataSet = ({
 	function updateDataSetItems(dataSetData) {
 		setItems(dataSetData.items);
 		setTotal(dataSetData.totalCount);
+
+		if (!dataSetData.items.length && dataSetData.totalCount > 0) {
+			setPageNumber(() => dataSetData.lastPage);
```

Comment:
@dsanz If we trust that every response returns `.lastPage` we are done

![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/ec447bc3-6697-42f4-ba8a-109cd322eed1)